packages/web3/src/entitlement.ts

```diff
@@ -494,7 +495,7 @@ export async function evaluateOperationsForEntitledWallet(
 export async function evaluateTree(
     controller: AbortController,
     linkedWallets: string[],
-    providers: ethers.providers.StaticJsonRpcProvider[],
+    providers: ethers.providers.BaseProvider[],
```

Comment:
I'm 100% for making this change to the BaseProvider so that a user can use whatever provider they want. 

I'm not excited about routing all of our calls through someone elses contract via an address that is updated via NPM

Can we deploy our own multi call contract and use that address with this library?

Reply:
We have our own MultiCall contract for this https://github.com/river-build/river/blob/main/contracts/src/diamond/utils/multicall/Multicall.sol

Reply:
0xsequence should be an established team. Unfortunately their [`MulticallProvider`](https://www.npmjs.com/package/@0xsequence/multicall) uses their own [`MultiCallUtils`](https://github.com/0xsequence/wallet-contracts/blob/master/contracts/modules/utils/MultiCallUtils.sol) which is deployed and verified [universally](https://blockscan.com/address/0xdbbFa3cB3B087B64F4ef5E3D20Dda2488AA244e6). We can deploy our own instance but it will still have to adhere to their interface.
If using viem multicall, the [`Multicall3`](https://www.multicall3.com/) contract is used.

Reply:
@shuhuiluo are you still going to pull the StaticJsonRpcProvider ==> BaseProvider change out of this pr and submit it? I'd love to see that change get in.

Reply:
Sure.